cli/commands/update.go

```diff
@@ -75,60 +75,56 @@ type UpdateRequest struct {
 	OptionsJSON    map[string]interface{} `json:"options,omitempty"`
 }
 
-func printStatus() {
-	// TODO: implement
-	client.LogMessage("Status has not been implemented yet. Please use `dcos %s --name=%s plan show` to view progress.", config.ModuleName, config.ServiceName)
-}
-
-func readJSONFile(filename string) (map[string]interface{}, error) {
-	fileBytes, err := ioutil.ReadFile(filename)
-	if err != nil {
-		return nil, err
-	}
-	return client.UnmarshalJSON(fileBytes)
-}
-
-func parseUpdateResponse(responseBytes []byte) (string, error) {
-	responseJSON, err := client.UnmarshalJSON(responseBytes)
-	if err != nil {
-		return "", err
-	}
-	return string(responseJSON["marathonDeploymentId"].(string)), nil
-}
-
 func doUpdate(optionsFile, packageVersion string) {
 	request := UpdateRequest{AppID: config.ServiceName}
 	if len(packageVersion) > 0 {
 		request.PackageVersion = packageVersion
 	}
 	if len(optionsFile) > 0 {
-		optionsJSON, err := readJSONFile(optionsFile)
+		fileBytes, err := ioutil.ReadFile(optionsFile)
 		if err != nil {
-			client.LogMessage("Failed to load specified options file %s: %s", optionsFile, err)
+			client.LogMessageAndExit("Failed to load specified options file %s: %s", optionsFile, err)
+			return
+		}
+		optionsJSON, err := client.UnmarshalJSON(fileBytes)
+		if err != nil {
+			client.LogMessageAndExit("Failed to parse JSON in specified options file %s: %s", optionsFile, err)
 			return
 		}
 		request.OptionsJSON = optionsJSON
 	}
 	requestContent, _ := json.Marshal(request)
 	response := client.HTTPCosmosPostJSON("update", string(requestContent))
 	responseBytes := client.GetResponseBytes(response)
-	// TODO: do something interesting with update response
-	_, err := parseUpdateResponse(responseBytes)
+	_, err := client.UnmarshalJSON(responseBytes)
 	if err != nil {
 		reportErrorAndExit(err, responseBytes)
 	}
-	client.LogMessage(fmt.Sprintf("Update started. Please use `dcos %s --name=%s service update --status` to view progress.", config.ModuleName, config.ServiceName))
+	client.LogMessage(
+		"Update started. Please use `dcos %s --name=%s service update status` to view progress.",
+		config.ModuleName, config.ServiceName)
 }
 
 func (cmd *UpdateHandler) UpdateConfiguration(c *kingpin.ParseContext) error {
-	if cmd.Status {
-		printStatus()
-		return nil
-	}
 	doUpdate(cmd.OptionsFile, cmd.PackageVersion)
 	return nil
 }
 
+func printStatus(rawJson bool) {
+	planName := "deploy"
+	response := client.HTTPServiceGet(fmt.Sprintf("v1/plans/%s", planName))
```

Comment:
We should add error handling for the 404 error case here (probably to `http.go`'s `checkHTTPResponse` method that behaves in the same way as for the 500 error case (i.e. calls `printServiceNameErrorAndExit`).

```
➜  helloworld git:(cli-plan-status) ✗ dcos hello-world update status
2017/06/02 17:17:48 Update started. Please use `dcos hello-world --name=hello-world service update status` to view progress.
2017/06/02 17:17:48 HTTP GET Query for http://sunil-51i-elasticl-otjoj8l120s3-916731190.us-west-2.elb.amazonaws.com/service/hello-world/v1/plans/deploy failed: 404 Not Found
```
(In my case, the service was still initialising.)